js/everything.js

```diff
@@ -186,157 +159,85 @@ var errorMessage = {
-        getYouTubeVideoDescription(
-            videoID,
-            youTubeDataApiKey,
-            function(data) {
-                if (data.items.length === 0) {
-                    errorMessage.show("Video description not found");
-                }
-                else {
-                    description = data.items[0].snippet.description;
-                }
-            },
-            function(jqXHR, textStatus, errorThrown) {
-                var responseText = JSON.parse(jqXHR.error().responseText);
-                errorMessage.show(responseText.error.errors[0].message);
-                console.log("Video Description error", errorThrown);
-            }
-        );
-    }
+var ZenPlayer = {
+    init: function(videoID) {
+        //Inject svg with control icons
+        $("#plyr-svg").load("../bower_components/plyr/dist/sprite.svg");
 
-    return description;
-}
+        plyrPlayer = document.querySelector(".plyr");
 
-function setupTitle(title, url) {
-    // Prepend music note only if title does not already begin with one.
-    var tmpVideoTitle = title;
-    if (!/^[\u2669\u266A\u266B\u266C\u266D\u266E\u266F]/.test(tmpVideoTitle)) {
-        tmpVideoTitle = "<i class=\"fa fa-music\"></i> " + tmpVideoTitle;
-    }
-    $("#zen-video-title").html(tmpVideoTitle);
-    $("#zen-video-title").attr("href", url);
-}
+        plyr.setup(plyrPlayer, {
+            autoplay: true,
+            controls:["play", "current-time", "duration", "mute", "volume"]
+        });
 
-function setupVideoDescription(videoDescription) {
-    var description = anchorURLs(videoDescription);
-    $("#zen-video-description").html(description);
-    $("#zen-video-description").hide();
+        //Load video into Plyr player
+        if (plyrPlayer.plyr) {
+            var that = this;
+            plyrPlayer.addEventListener("error", function(event) {
+                console.log("error");
+                console.log(event);
+                console.log(event.target.plyr);
+                console.log(event.target.plyr.embed.api);
+            });
 
-    $("#toggleDescription").click(function(event) {
-        event.preventDefault();
+            plyrPlayer.addEventListener("ready", function() {
+                // Gather video info
+                that.videoTitle = plyrPlayer.plyr.embed.getVideoData().title;
+                that.videoAuthor = plyrPlayer.plyr.embed.getVideoData().author;
+                that.videoDuration = plyrPlayer.plyr.embed.getDuration();
+                that.videoDescription = that.getVideoDescription(videoID);
+                that.videoUrl = plyrPlayer.plyr.embed.getVideoUrl();
+
+                // Initialize UI
+                that.setupTitle();
+                that.setupVideoDescription();
+                that.setupPlyrToggle();
+
+                // Start video from where we left off
+                // TODO: plyr should already implement that, verify
+                // plyrPlayer.plyr.embed.seekTo(loadTime());
```

Comment:
Done, I also did some magic so it looks like you committed the change :sparkles: